packages/react-components/react-tags/src/components/TagButton/useTagButton.tsx

```diff
@@ -74,13 +102,13 @@ export const useTagButton_unstable = (props: TagButtonProps, ref: React.Ref<HTML
     icon: resolveShorthand(props.icon),
     primaryText: resolveShorthand(props.primaryText, { required: true }),
     secondaryText: resolveShorthand(props.secondaryText),
-    dismissButton: useARIAButtonShorthand(props.dismissButton, {
-      required: props.dismissible,
-      defaultProps: {
-        disabled,
-        type: 'button',
-        children: <DismissIcon />,
-      },
-    }),
+
+    dismissButton: dismissButtonShorthand
```

Comment:
why is this ternary statement necessary? Is it possible for `useARIAButtonShorthand` to return undefined/null?

Also you should be able to pass event listeners to `useARIAButtonShorthand` directly and itwill do the merging

Reply:
`dismissButton` slot is not using `ARIAButtonSlotProps` (is just `Slot<'button'>, I see no reason to even use `useARIAButtonShorthand` internally if for the moment we're not supporting the types that benefit from it.

Reply:
@ling1726 the ternary is for styling - I need dismissButton to be `undefined` when it is not specified by the user. So I can't do: 
```
dismissButton: {
          ...dismissButtonShorthand,
          onClick: onDismissButtonClick,
          onKeyDown: onDismissButtonKeyDown,
}
```
When there's dismissButton, content should have sharp border radius topRight/bottomRight. Therefore I'm doing `state.dismissButton && styles.dismissibleContent` in styles. 

And @bsunderhus I'm changing `useARIAButtonShorthand` to just `resolveShorthand` for dismissButton